test/modules/carto/layers/h3-tile-layer.spec.js

```diff
@@ -0,0 +1,92 @@
+import {h3GetResolution, h3ToChildren} from 'h3-js';
+import test from 'tape-promise/tape';
+import {generateLayerTests, testLayerAsync} from '@deck.gl/test-utils';
+import {_H3TileLayer as H3TileLayer} from '@deck.gl/carto';
+import {WebMercatorViewport} from '@deck.gl/core';
+import {testPickingLayer} from '../../layers/test-picking-layer';
+
+const TILES = ['https://tile-url.com/{i}'];
+const TILEJSON = {
+  maxresolution: 10,
+  tiles: TILES
+};
+
+test('H3TileLayer', async t => {
+  const testCases = generateLayerTests({
+    Layer: H3TileLayer,
+    assert: t.ok,
+    onBeforeUpdate: ({testCase}) => t.comment(testCase.title)
+  });
+  await testLayerAsync({Layer: H3TileLayer, testCases, onError: t.notOk});
+  t.end();
+});
+
+test('H3TileLayer tilejson', async t => {
+  const testCases = [
+    {
+      Layer: H3TileLayer,
+      props: {
+        data: TILEJSON,
+        getTileData: () => []
+      },
+      assert: t.ok,
+      onAfterUpdate({layer, subLayers}) {
+        if (!layer.isLoaded) {
+          t.equal(subLayers.length, 1, 'Rendered sublayers');
+          t.deepEqual(subLayers[0].props.data, TILES, 'Extract tiles from tilejson');
+          t.deepEqual(subLayers[0].props.maxZoom, 10, 'Extract maxZoom from tilejson');
```

Comment:
I'm confused here, in h3 maxZoom should be the zoom that matches with the resolution 10.

Reply:
Just realized that backend already does the conversion

Reply:
Yes it is a little messy with the variable names, as the base `Tileset2D` class uses `maxZoom` - even if it treats it as an abstract "resolution" parameter